pkg/controller/privatelink/actuator/awsactuator/awshubactuator.go

```diff
@@ -0,0 +1,593 @@
+package awsactuator
+
+import (
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/route53"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/hive/pkg/awsclient"
+	"github.com/openshift/hive/pkg/controller/privatelink/actuator"
+	"github.com/openshift/hive/pkg/controller/privatelink/conditions"
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
+)
+
+var (
+	errNoHostedZoneFoundForVPC = errors.New("no hosted zone found")
+)
+
+// Ensure AWSHubActuator implements the Actuator interface. This will fail at compile time when false.
+var _ actuator.Actuator = &AWSHubActuator{}
+
+type AWSHubActuator struct {
+	client *client.Client
+
+	config *hivev1.AWSPrivateLinkConfig
+
+	// testing purpose
+	awsClientFn awsClientFn
+
+	awsClientHub awsclient.Client
+}
+
+// New creates a new AWS Hub Actuator
+func NewAWSHubActuator(
+	client *client.Client,
+	config *hivev1.AWSPrivateLinkConfig,
+	awsClientFn awsClientFn,
+	logger log.FieldLogger) (*AWSHubActuator, error) {
```

Comment:
Is this really how go fmt decided to format this? Not a fan

Reply:
I'm not a huge fan either. It's a mix of go fmt and my own attempt at "each on their own line so we don't have a super long line."  I messed with muti-line usage in a few places, reverting most of them back to a single line. I think this one had a lot more parameters and isn't so bad trimmed down.

Another possible way to do this is similar to:
```
func NewAWSHubActuator(
	client *client.Client,
	config *hivev1.AWSPrivateLinkConfig,
	awsClientFn awsClientFn,
	logger log.FieldLogger,
) (*AWSHubActuator, error) {
```

Reply:
I would prefer that version personally.  I like putting parameters on their own line because it becomes a lot easier to read diffs when parameters are changed or reordered

Reply:
FYI: The urge to break up long lines is an artifact of my (probably oudated) training to keep every line under 80 characters so that the code is maintainable in a text editor within a *nix shell terminal.

Reply:
If there are any significant changes required, I'll also see about cleaning some of these up.